src/FriendshipButtons.tsx

```diff
@@ -0,0 +1,187 @@
+      },
+      body: JSON.stringify({
+        id: friend.id,
+        action: "accept",
+      }),
+    })
+      .then((response) => {
+        if (response.ok) {
+          setUsers(
+            users.map((user) =>
+              user.id !== friend.id ? user : { ...friend, status: "friends" }
+            )
+          );
+          addToast({
+            header: "Friend added!",
+            body: `You are now friends with ${friend.name}`,
+            type: "success",
+          });
+        } else {
+          throw Error;
+        }
+      })
+      .catch((error) => {
+        console.error(error);
+        addToast({
+          header: "Could not add friend",
+          body: "Something went wrong. Please try again in a few moments",
+          type: "error",
+        });
+      });
+  }, [friend, addToast, users, setUsers]);
+
+  return (
+    <Button variant="primary" onClick={acceptFriendship} size="sm" {...props}>
+      {children ?? "Accept"}
+    </Button>
+  );
+};
+
+interface RejectButtonProps {
+  gated?: boolean;
+}
+
+export const FriendRejectButton: React.FC<
+  FriendButtonProps & RejectButtonProps
+> = ({ friend, children, gated, ...props }) => {
+  const { addToast } = useToasts();
+  const { users, setUsers } = useUserList();
+  const requestConfirm = useContext(ConfirmationContext);
+
+  const deleteFriendship = useCallback(
+    () =>
+      fetch(`http://paralibrary.digital/api/friends/${friend.id}`, {
+        method: "POST",
+        credentials: "include",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          id: friend.id,
+          action: "reject",
+        }),
+      })
+        .then((response) => {
+          if (response.ok) {
+            setUsers(
+              users.map((user) =>
+                user.id !== friend.id ? user : { ...friend, status: null }
+              )
+            );
+          } else {
+            throw Error;
+          }
+        })
+        .catch((error) => {
+          console.error(error);
+          addToast({
+            header: "Action could not be completed",
+            body: "Something went wrong. Please try again in a few moments",
+            type: "error",
+          });
+        }),
+    [friend, addToast, users, setUsers]
+  );
+
+  const gatedDeletion = useCallback(() => {
+    requestConfirm(deleteFriendship, {
+      title: `Unfriend ${friend.name}?`,
+      body:
+        'Are you sure you want to remove this user as a friend? You will no longer be able to loan books marked as "friends" or view certain books in each others\' libraries.',
+    });
+  }, [deleteFriendship, friend, requestConfirm]);
+
+  return (
+    <Button
+      variant="outline-danger"
+      onClick={gated ? gatedDeletion : deleteFriendship}
+      size="sm"
+      {...props}
+    >
+      {children ?? "Reject"}
+    </Button>
+  );
+};
+
+export const FriendAcceptReject: React.FC<FriendButtonProps> = ({ friend }) => {
```

Comment:
This components seems fairly superfluous to me. It's only being used in one place that I can see and it's just a short cut for adding both buttons since you aren't wrapping them in any styles.
It seems to me you could just put these two buttons separately where they are needed. I get how they are structurally linked though so that's perhaps reason to keep it.